app/models/throttle.rb

```diff
@@ -48,10 +49,51 @@ class Throttle < ApplicationRecord
     },
   }.freeze
 
+  # @param [User,Integer,String] target User, its ID, or a string identifier
+  # @param [Symbol] throttle_type
+  # @return [Throttle]
+  def self.for(target:, throttle_type:)
+    throttle = case target
+    when User
+      find_or_create_by(user: target, throttle_type: throttle_type)
+    when Integer
+      find_or_create_by(user_id: target, throttle_type: throttle_type)
+    when String
+      find_or_create_by(target: target, throttle_type: throttle_type)
+    else
+      raise "Unknown throttle target class=#{target.class}"
+    end
+
+    throttle.tap do |t|
+      t.reset_if_expired_and_maxed
+    end
+  end
+
+  # @return [Throttle]
+  def increment
+    return self if maxed?
+    update(attempts: attempts + 1, attempted_at: Time.zone.now)
+    self
+  end
+
   def throttled?
     !expired? && maxed?
   end
 
+  def throttled_else_increment?
```

Comment:
Open to new name suggestions here! It's not a normal predicate method because it has a side effect. Maybe just drop the `?` at the end? 😬